### DISCUSSION TITLE
How get replicators errors

### DISCUSSION THREAD

**Discussion by filipenanclarez** [cofu-app/cbl-dart#423] [Q&A] [answered]
When replicator crash ( like #387 ), the replicator stop and i can get this event on `if (change.status.activity == cbl.ReplicatorActivityLevel.stopped) {`
of replicator change listener `await cblReplicator!.addChangeListener((change) async { ... }` 

When this happen, how can i get the error messages ( same messages showed in debug console ) ? 

I want send this messages to crashlytcs.

**Comment by blaugold** [comment, marked as answer]
If I understand it correctly, the information in [`ReplicatorStatus.error`](https://pub.dev/documentation/cbl/latest/cbl/ReplicatorStatus/error.html) is not sufficient, and you would like to capture related log messages as well.

There is no way to directly access those log messages because I don't have the ability to tie them to a specific replicator change. 

I would suggest listening to log messages, optionally filter out those you are not interested in, store them in a ring buffer that has a fixed length (e.g. 200) and send the messages in the buffer to crashlytics when some event or error occurs for which you want to have more information. I'm not exactly sure crashlytics offers to that effect. 

For Sentry, there is an integration that does exactly that for you: [`cbl_sentry`](https://pub.dev/packages/cbl_sentry).

Some day there might be a similar integration for crashlytics, but I have not looked into it.

**Reply by blaugold** [reply]
P.S. If you decide to use `cbl_sentry`, errors from a replicator don't throw, so they don't trigger a crash report automatically. You can manually do that, though:

```dart
if (change.status.error != null) {
  Sentry.captureException(change.status.error!);
}
```

**Reply by filipenanclarez** [reply]
Thank you. The `ReplicatorStatus.error` is the trick.
